bot/exts/info/subscribe.py

```diff
@@ -0,0 +1,149 @@
+            await interaction.response.send_message(
+                ":x: This is not your command to react to!",
+                ephemeral=True
+            )
+            return False
+        return True
+
+
+class SingleRoleButton(discord.ui.Button):
+    """A button that adds or removes a role from the member depending on it's current state."""
+
+    ADD_STYLE = discord.ButtonStyle.success
+    REMOVE_STYLE = discord.ButtonStyle.secondary
+    LABEL_FORMAT = "{action} role {role_name}"
+    CUSTOM_ID_FORMAT = "subscribe-{role_id}"
+
+    def __init__(self, role: discord.Role, assigned: bool, row: int):
+        super().__init__(
+            style=self.REMOVE_STYLE if assigned else self.ADD_STYLE,
+            label=self.LABEL_FORMAT.format(action="Remove" if assigned else "Add", role_name=role.name),
+            custom_id=self.CUSTOM_ID_FORMAT.format(role_id=role.id),
+            row=row,
+        )
+        self.role = role
+        self.assigned = assigned
+
+    async def callback(self, interaction: Interaction) -> None:
+        """Update the member's role and change button text to reflect current text."""
+        if isinstance(interaction.user, discord.User):
+            await interaction.message.delete()
+            self.view.stop()
+            return
+
+        await members.handle_role_change(
+            interaction.user,
+            interaction.user.remove_roles if self.assigned else interaction.user.add_roles,
+            self.role,
+        )
+
+        self.assigned = not self.assigned
+        await self.update_view(interaction)
+        await interaction.response.send_message(
+            self.LABEL_FORMAT.format(action="Added" if self.assigned else "Removed", role_name=self.role.name),
+            ephemeral=True,
+        )
+
+    async def update_view(self, interaction: Interaction) -> None:
+        """Updates the original interaction message with a new view object with the updated buttons."""
+        self.style = self.REMOVE_STYLE if self.assigned else self.ADD_STYLE
+        self.label = self.LABEL_FORMAT.format(action="Remove" if self.assigned else "Add", role_name=self.role.name)
+        view_list = [
+            self if button.custom_id == self.custom_id else button
+            for button in self.view.children
+        ]
+        self.view.children = view_list
+        try:
+            await interaction.message.edit(view=self.view)
+        except discord.NotFound:
+            log.debug("Subscribe message for %s removed before buttons could be updated", interaction.user)
+            self.view.stop()
+
+
+class Subscribe(commands.Cog):
+    """Cog to allow user to self-assign & remove the roles present in ASSIGNABLE_ROLES."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.init_task = scheduling.create_task(self.init_cog(), event_loop=self.bot.loop)
+        self.assignable_roles: list[discord.Role] = []
+        self.guild = discord.Guild = None
```

Comment:
```suggestion
        self.guild: discord.Guild = None
```
Did you mean to do this?

Reply:
yes

Reply:
https://github.com/python-discord/bot/pull/1868/commits/44f47fb45d2dd7aaa47b30d2164b659da0d6ae95